src/Middleware/JsonBodyParser.php

```diff
@@ -64,15 +63,9 @@ public function withThrowException(bool $value): self
      */
     private function parse(string $body)
     {
-        try {
-            $result = json_decode($body, $this->assoc, $this->depth, $this->options);
-            if (is_array($result) || is_object($result)) {
-                return $result;
-            }
-        } catch (\JsonException $e) {
-            if ($this->throwException) {
-                throw new \InvalidArgumentException('Invalid JSON data in request body: ' . $e->getMessage());
-            }
+        $result = json_decode($body, $this->assoc, $this->depth, $this->options);
+        if (is_array($result) || is_object($result)) {
+            return $result;
         }
         return null;
```

Comment:
```suggestion
        if (false === $result) {
            return null;
        }
        return $result;
```

Reply:
With such a check, this will pass https://3v4l.org/9e159
`withParsedBody` accepts only `array|object|null`

Reply:
https://tools.ietf.org/html/rfc7159#section-13
Strings, numbers and booleans are valid values.
Use https://www.php.net/manual/ru/function.json-last-error.php to detect the decoding error

Reply:
> https://tools.ietf.org/html/rfc7159#section-13
> Strings, numbers and booleans are valid values.
> Use https://www.php.net/manual/ru/function.json-last-error.php to detect the decoding error

but `withParsedBody` not supported other data type